diff --git i/lua/nui/popup/border.lua w/lua/nui/popup/border.lua
index 0160a49..805a779 100644
--- i/lua/nui/popup/border.lua
+++ w/lua/nui/popup/border.lua
@@ -538,7 +538,7 @@ function Border:_relayout()
   internal.lines = calculate_buf_lines(internal)
 
   if self.winid then
-    vim.api.nvim_win_set_config(self.winid, self.win_config)
+    pcall(vim.api.nvim_win_set_config, { self.winid, self.win_config })
   end
 
   if self.bufnr then
diff --git i/lua/nui/popup/init.lua w/lua/nui/popup/init.lua
index fe838c7..3da53bd 100644
--- i/lua/nui/popup/init.lua
+++ w/lua/nui/popup/init.lua
@@ -361,7 +361,7 @@ function Popup:update_layout(config)
   self._.layout_ready = true
 
   if self.winid then
-    vim.api.nvim_win_set_config(self.winid, self.win_config)
+    pcall(vim.api.nvim_win_set_config, { self.winid, self.win_config })
   end
 end
 
